index.d.ts

```diff
@@ -487,6 +488,18 @@ declare namespace mercurius {
      * receive an array of responses within a single request.
      */
     allowBatchedQueries?: boolean;
+
+    /**
+     * Customize the graphql routers initialized by mercurius with 
+     * more fastify route options. 
+     * 
+     * Usecases:
+     * - Add more validation 
+     * - change the schema structure 
+     * - Hook on the request's life cycle
+     * - Increase body size limit for larger queries
+    */
+    additionalRouteProps?:Omit<RouteOptions,"handler" | "wsHandler">
```

Comment:
let's add types tests for this

Reply:
There's no type tests for the options in place though, do we want it to be a separate PR for all of them ?

Reply:
I mean, just add a test which checks that you can supply route options (and which ones perhaps) here https://github.com/mercurius-js/mercurius/blob/master/test/types/index.ts